lib/android/app/src/main/java/com/reactnativenavigation/utils/ViewExtensions.kt

```diff
@@ -3,7 +3,7 @@ package com.reactnativenavigation.utils
 import android.view.View
 import android.view.ViewManager
 
-fun View.removeFromParent() = (this.parent as? ViewManager)?.removeView(this)
+fun View?.removeFromParent() = (this?.parent as? ViewManager)?.removeView(this)
```

Comment:
I don't understand this change. @swabbass can you please explain it to me? Wouldn't you rather have the null check on the actual receiver? Hiding it makes the code less explicit.

Reply:
This is mainly changed due to an error that happened while compiling tests that check null views.

Since `View?` can receive null or non-null values and in Kotlin it is treated as a different type,  we can run extensions declared for this type (`View?`)  only, and check the receiver for null and handling it internally, in our code the `?` operator behaves as a null check.

It seems that hiding it would make the code less explicit, but, since the declared view is nullable in java by default and in Kotlin we use `View?`,  it's understandable that accessing it will need a null check, that means it demands devs attention to check for null, what I did for one extension is to make the check internally since the types declared explicitly as nullable, makes the code cleaner with short single liners.

Reply:
But with null coalescing, our code is already cleaner. Even the motivation here is wrong - this change was introduced due to compilation errors in tests. You shouldn't change "production code" because of tests.

Can you call `addView()` on a null ViewGroup? Of course not, it makes no sense. `(null as View?).removeFromParent()` also makes no sense. What's even worst, is that this change might prevent us from detecting actual bugs early on in the development cycle.

For example, if you're trying to remove a view from its parent, but for some reason it's null. Instead of crashing during runtime, your code will work.

Reply:
Production code did not change its the same thing, as it was in java and I moved it to Extension i forgot to add ? and that gave us the error.

Extensions and member functions are totally different in concept, `addView` cant be called on null, but extensions can!, [see](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-null-or-empty.html) `String.IsNullOrEmpty()` for nullable types we can use nullable syntax, in Java mindset it is a static function that receives the object with nullable annotation and its called this so we need to check nulls if the value can have null, otherwise it's safe.

About the example, if I'm trying to remove the view from its parent and the view is null then nothing will happen simply the same behaviour that was before, which was checking nulls before removing the view,  [see](https://github.com/wix/react-native-navigation/blob/b53f13c60fe6498ad1690adf1dfd7509f602965c/lib/android/app/src/main/java/com/reactnativenavigation/utils/ViewUtils.java#L118). 

What I'm trying to say here, it's a different syntax but the same thing happens in Java and Kotlin implementations.
Kotlin and Java are different mindset languages that turn out to be JVM languages.